File: car-rental-backend/src/main/java/com/carrental/car_rental_backend/service/CarService.java
package com.carrental.car_rental_backend.service;

import java.util.List;

import com.carrental.car_rental_backend.entity.Car;

public interface CarService {
    List<Car> getAllCars();

    Car getCarById(int id);

    Car saveCar(Car car);

    Car updateCar(int id, Car car);

    void deleteCar(int id);
}

File: car-rental-frontend/frontend/src/services/carService.js
import axios from "axios"

const BASE_URL = "http://localhost:8080"

export const getAllCars = () => {
    return axios.get(`${BASE_URL}/cars`)
}

export const getCarById = (id) => {
    return axios.get(`${BASE_URL}/cars/${id}`)
}

export const addCar = (car) => {
    return axios.post(`${BASE_URL}/cars`, car)
}

export const updateCar = (id, car) => {
    return axios.put(`${BASE_URL}/cars/${id}`, car)
}

export const deleteCar = (id) => {
    return axios.delete(`${BASE_URL}/cars/${id}`)
}